p4-16/spec/P4-16-spec.adoc

```diff
@@ -3657,21 +3663,8 @@ The `int<W>` datatype also support the following operations:
 ** all result bits are zero when shifting a non-negative value right
 ** all result bits are one when shifting a negative value right
 * Extraction of a set of contiguous bits, also known as a slice,
-  denoted by `[H:L]`, where `H` and `L` must be expressions that evaluate to
-  non-negative, local compile-time known values, and `H >= L` must be true.
-  The types of `H` and `L` (which do not need to be identical)
-  must be numeric (<<sec-numeric-values>>).
-  The result is an unsigned bit-string of width `H - L + 1`, including the bits
-  numbered from `L` (which becomes the least significant bit of the result)
-  to `H` (the most significant bit of the result) from the source operand.
-  The conditions `+0 <= L <= H < W+` are checked statically (where `W` is
-  the length of the source bit-string). Note that both endpoints of
-  the extraction are inclusive. The bounds are required to be values
-  that are known at compile time so that the width of the result can be
-  computed at compile time. Slices are also l-values, which means that P4
-  supports assigning to a slice: `e[H:L] = x`.
-  The effect of this statement is to set bits `H` through `L` of `e` to the
-  bit-pattern represented by `x`, and leaves all other bits of `e` unchanged.
+  denoted by `[H:L]` or `[L+:W]`, with the same semantics as slices of
+  bit strings.
```

Comment:
I think we need to explicitly state that the result is unsigned. One could wonder what happens if the slice starts at the MSB.

Reply:
Yes, the result of a slice should always be unsigned (`bit<n>`, never `int<n>`)